applications/openshift/kubelet/kubelet_protect_kernel_defaults/rule.yml

```diff
@@ -0,0 +1,41 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: 'Kubelet - Disable Kernel Default Protection'
+
+description: |-
+    By default, the kubelet can adjust kernel parameters automatically.
+    If kernel default protection is enabled, the kubelet will instead
+    error if kernel parameters do not match what it expects.
```

Comment:
Question: If for some reason, it is set to True, how can I change this?

Reply:
It's enforced by the platform, so unless there is an exposed knob there is not a persistent way to change it.

Reply:
If it is enforced by the platform and cannot be changed say on install, update, etc, is there an actual reason to have this rule at all?

Reply:
Perhaps not? It's still a valid option to the kubelet, but it should never appear in practice. There are other rules where we check for essentially deprecated flags

Reply:
it's in the benchmark... I guess the intent is to not have folks wreck their deployments thinking it'll bring them extra security.

Reply:
This is the question. Can you actually configure it in OCP. If so, yes this rule should be there. If there is no way, adding this rule makes no sense.

Reply:
It's possible to configure as an "unsupportedOverride".

Reply:
oh... actually this isn't even an unsupported override. This is settable and even supported via https://docs.openshift.com/container-platform/4.6/nodes/nodes/nodes-nodes-managing.html (it allows you to set almost any setting for the kubelet).

Reply:
So yeah, it's relevant for us to check this.

Reply:
We tested this in my local deployment and it makes the kubelet crash...

Reply:
Upon further investigation, it seems that the arguments that the kubelet needs to set are not many... maybe we can fix this outside of the kubelet and disallow it from changing this.

Reply:
@JAORMX it would be best to support preventing the DoS and the kubelet from making changes to the nodes. Otherwise, we open a vulnerability to close a vulnerability so to speak. If we can fix this outside of the kubelet, that would be amazing.

Reply:
@redhatrises researching this. I'll probably come back with results later today.